### DISCUSSION TITLE
CLEAN_EXCLUDE add regex or only affect TARGET_FOLDER?

### DISCUSSION THREAD

**Discussion by jerdra** [JamesIves/github-pages-deploy-action#557] [Q&A]
Hey!

I have a setup where I maintain multiple versions of my documentation on github pages (one for each tag, these appear as directories named after their respective tags in my gh-pages root directory). These are set to update whenever a new tag is released using the `CLEAN: true` setting for this action. This part works as intended. 

However I also plan to maintain a `dev` version (in a `dev` directory) which updates on `push` actions to my `dev` branch. I can't set `CLEAN: true` here since it would wipe out my tagged documentation. Ideally, I'd like to update the `dev` directory but only after cleaning it out, to ensure that we don't have old pages lying around within the directory. Is there a solution for these sorts of setups? Two ways that I can think of to deal with these scenarios are:

1. `CLEAN_EXCLUDE` takes a regex, this would be more flexible that listing each of the directories to maintain
2. `CLEAN` only affects `TARGET_FOLDER` when both are used in combination

Are these already implemented features (i can't find info on a regex, or special behaviours w/TARGET_FOLDER)? I can imagine this being useful when you want to update only a subset of the gh-pages branch, let me know what you think!

**Comment by jerdra** [comment]
Just tested the latter suggestion (CLEAN + TARGET_FOLDER) and it looks like it works as I'd like it to! Should this behaviour be documented in the README perhaps?

**Comment by Pike** [comment]
Yeah, updating the docs sounds good to me.

The docs for `clean` need to be updated anyway, #373 accidentally(?) started documenting that option to be on by default, which it isn't. AFAICT.

**Reply by Pike** [reply]
So, `clean` is true by default, but the docs weren't consistent about that. I've taken a stab at making that consistent in #559

**Reply by jerdra** [reply]
thanks for setting this up @Pike. One very minor comment is that it might provide more clarity to explicitly state that `clean` will examine the folder specified by the `target-folder` option if it is set. Otherwise it will examine the default deployment directory (root directory of gh-pages branch).

I think you already refer to a "target folder", but having the if-else clause would make it very explicit how that is determined
